### PR TITLE
Use default `model_name` in `metadata_update`

### PR DESCRIPTION
**Pull request by lvwerra**
As reported by a user, the `metadata_update` does not work anymore (likely introduced in #940) if no `"name"` field in the `"model-index"` is provided. This breaks for example `evaluate.push_to_hub` where evaluation results are pushed to the hub. 

With this PR the following behaviour is supported:
- if `"model-index"` of new metadata has a `"name"` it is used (1)
- if there is no `"name"`
    - if the existing repo card has one use this one (2)
    - else use the repo name as default (3)

Since the `"name"` is I think a little known feature of the PwC integration and providing a model name when pushing the results to a model repo seems a bit redundant I feel like having good defaults and not bother the user with it is a good idea. If you think this is a bad idea, we can throw an error in case (3).

cc @julien-c

### CODE REVIEW

**Comment by Wauplin** [comment]
> I think a little known feature (...) I feel like having good defaults and not bother the user with it is a good idea.

To be honest, I don't know enough about metadata to really have an opinion on the "name" under "model-index". But I agree with you that in general we don't want to bother users about very specific features. I ping @nateraw would worked on that and might be more opinionated.  

---

Otherwise, I noticed that for (3) you are talking about the repo name (e.g. "bart-base") and not the repo id (e.g. "facebook/bart-base"). I guess that's intended but wanted to highlight it for others in the discussion.

And also if we move forward on this PR, @lvwerra  could you please add at least one test that triggers case (2) or (3) (e.g. a case that is currently broken but that your PR solves). Thanks in advance !

**Comment by lvwerra** [comment]
Yes, will add tests if people are happy with the suggested change, thanks @Wauplin!

File: src/huggingface_hub/repocard.py
Comment (julien-c) [inline src/huggingface_hub/repocard.py:778]:
```suggestion
                value[0]["name"] = getattr(card, "model_name", repo_id)
```

**Review by julien-c — COMMENTED** [review]
personally I would use the full repo_id, but otherwise, lgtm!

**Comment by lvwerra** [comment]
Sounds good, also added some tests :)

**Review by Wauplin — APPROVED** [review]
Looks good to me, thanks for adding the test !
Please wait for @nateraw 's review before merging :)

**Review by nateraw — APPROVED** [review]
LGTM thanks for the update. One concern is that we do mention this required name in the docs for model card. 

Do we need to update the [docstring](https://huggingface.co/docs/huggingface_hub/package_reference/cards#huggingface_hub.ModelCardData.model_name) around this? Mentioned in [eval_result](https://huggingface.co/docs/huggingface_hub/package_reference/cards#huggingface_hub.ModelCardData.eval_results) kwarg docstring as well.

**Comment by nateraw** [comment]
Thanks for docstring update. ❤️ 

once quality tests/other tests pass this is good to go I think :)

**Comment by Wauplin** [comment]
I'm finally merging the PR. Thanks @lvwerra for all the modifications ! :)

**Pull request merged**
